corgi/core/models.py

```diff
@@ -923,11 +923,11 @@ def get_provides(self, include_dev=True):
         type_list = [ComponentNode.ComponentNodeType.PROVIDES]
         if include_dev:
             type_list.append(ComponentNode.ComponentNodeType.PROVIDES_DEV)
-        first_parent = self.cnodes.get_queryset().first()
-        if first_parent.get_descendant_count() == 0:
-            return []
-        for descendant in first_parent.get_descendants().filter(type__in=type_list):
-            provides.add(descendant.purl)
+        for cnode in self.cnodes.get_queryset():
+            if cnode.get_descendant_count() == 0:
+                break
```

Comment:
Should this be "continue" instead of "break"? Even if one cnode doesn't have any descendants, we still want to find descendants with PROVIDES type for other cnodes, right?

Reply:
Good catch, it should be continue so I changed it. I left the check of get_descendant_count because it avoids a DB call, see https://django-mptt.readthedocs.io/en/latest/models.html#get-descendant-count